### PR TITLE
docs: Add usage information to `HttpInterceptorFn`

### PR DESCRIPTION
**Pull request by JeanMeche**
Simple doc addendum

Edit: [Direct link](https://ng-dev-previews-fw--pr-angular-angular-51236-64x6v904.web.app/api/common/http/HttpInterceptorFn)

### CODE REVIEW

**Comment by JeanMeche** [comment]
The failed internal tests are likely unrelated.

File: packages/common/http/src/interceptor.ts [hunk 1]
```diff
@@ -85,8 +86,33 @@ export type HttpHandlerFn = (req: HttpRequest<unknown>) => Observable<HttpEvent<
...
- * `HttpInterceptorFn`s have access to `inject()` via the `EnvironmentInjector` from which they were
- * configured.
+ * `HttpInterceptorFn`s are executed in an [injection context](/guide/dependency-injection-context).
+ * They have access to `inject()` via the `EnvironmentInjector` from which they were configured.
+ *
+ * @see [HTTP Guide](guide/http-intercept-requests-and-responses)
+ * @see {@link withInterceptors}
+ *
+ * @usageNotes
+ * Here is a do-nothing noop interceptor that passes the request through without touching it:
```
Comment (AndrewKushnir) [inline packages/common/http/src/interceptor.ts:96]:
```suggestion
 * Here is a noop interceptor that passes the request through without modifying it:
```

File: packages/common/http/src/interceptor.ts [hunk 2]
```diff
@@ -85,8 +86,33 @@ export type HttpHandlerFn = (req: HttpRequest<unknown>) => Observable<HttpEvent<
...
+ * ```typescript
+ * export const noopInterceptor: HttpInterceptorFn = (req: HttpRequest<unknown>, next:
+ * HttpHandlerFn) => { return next(modifiedReq);
+ * };
```
Comment (AndrewKushnir) [inline packages/common/http/src/interceptor.ts:100]: The formatting in this example (and the example below) seems a bit incorrect. Can we improve the readability a bit by splitting this code into a couple lines?
```suggestion
 * export const noopInterceptor: HttpInterceptorFn = (req: HttpRequest<unknown>, next:
 * HttpHandlerFn) => { return next(modifiedReq);
 * };
```

File: packages/common/http/src/interceptor.ts [hunk 3]
```diff
@@ -85,8 +86,33 @@ export type HttpHandlerFn = (req: HttpRequest<unknown>) => Observable<HttpEvent<
...
+ * ```
+ *
+ * If you must alter a request, clone it first and modify the clone before passing it to `next()`
```
Comment (AndrewKushnir) [inline packages/common/http/src/interceptor.ts:103]:
```suggestion
 * If you want to alter a request, clone it first and modify the clone before passing it to the `next()`
```

**Comment by AndrewKushnir** [comment]
@JeanMeche thanks for the PR, this is a great addition to the docs 👍

**Review by AndrewKushnir — APPROVED** [review]

**Pull request closed without merging**